app/frontend/src/components/Map.tsx

```diff
@@ -46,13 +47,33 @@ export default function Map({
 
   const containerRef = useRef<HTMLDivElement>(null);
 
+  /*
+  Allows sending cookies (counted as sensitive "credentials") on cross-origin requests when we grab GeoJSON/other data from the API.
+
+  Those APIs will return an error if the session cookie is not set as these APIs are secure and not public.
+  */
+  const transformRequest = (
+    url: string,
+    resourceType: ResourceType
+  ): RequestParameters => {
+    if (url.startsWith(URL)) {
+      return {
+        url: url,
+        credentials: "include",
+      };
+    }
+    return { url: url };
+  };
+
   useEffect(() => {
     if (!containerRef.current) return;
     const map = new mapboxgl.Map({
       container: containerRef.current,
       style: "mapbox://styles/mapbox/streets-v11",
       center: initialCenter,
       zoom: initialZoom,
+      hash: "map",
```

Comment:
This adds a `map` fragment to the URL to make it something like this `http://localhost:3000/map#map=0.57/0/-9.2`, so that when you go back, it'll remember where the map was and won't go back to the initial center

Reply:
Mega ultra nitpick, can't you change it to "view" or "location" or something so the url isn't `app.couchers.org/map#map=z/y/x`